.github/workflows/ci.yml

```
@@ -31,9 +31,11 @@ jobs:
       - run: npm run test
       - run: npm run e2e-test
         env: 
-          REFRESH_TOKEN: ${{ secrets.REFRESH_TOKEN }}
-          CLIENT_ID: ${{ secrets.CLIENT_ID }}
-          CLIENT_SECRET: ${{ secrets.CLIENT_SECRET }}
+          E2E_TEST_REFRESH_TOKEN: ${{ secrets.E2E_TEST_REFRESH_TOKEN }}
+          E2E_TEST_CLIENT_ID: ${{ secrets.E2E_TEST_CLIENT_ID }}
+          E2E_TEST_CLIENT_SECRET: ${{ secrets.E2E_TEST_CLIENT_SECRET }}
+          E2E_TEST_IDP_URL: ${{ secrets.E2E_TEST_IDP_URL }}
+          E2E_TEST_ESS_POD: ${{ secrets.E2E_TEST_ESS_POD }}
```

We can discuss this if you like, but the idea here is that this refers to a specific instance of an ESS deployment, and 'POD' is not really an instance (although it is kinda the name of the instance in the URL itself). i.e. we could also refer to Dev and Demo and Staging instances, and we'd name them similarly, i.e. DEV, DEMO, STAGING:
```suggestion
          E2E_TEST_ESS_PROD: ${{ secrets.E2E_TEST_ESS_PROD }}
```

---

Yeah, I wasn't quite sure yet what the best approach is. In solid-client I actually have the GitHub secrets use `PROD` in their name (so it's `E2E_TEST_ESS_PROD_POD`), which are then mapped to `E2E_TEST_ESS_POD` by the CI script. That way, we can store credentials for different environments in the secrets, and the CI can just run the e2e tests multiple times with different environment variables if it wants to test against different environments. Additionally, this would allow e.g. the ESS team to run the tests for one specific environment only after a deploy of that environment.

However, I'd also be fine with the e2e test script just listing all different environments themselves. As above, I didn't do that for this PR to limit the scope.

(That said, I'd be wary to test against dev environments of ESS - if every breakage there slows us down, that might be a bit too much.)

---

Yeah, good point. I've noted this aspect of the issue in the new 'update naming' ticket I just created. Having all these hard-coded, and manually updated, GitHub secrets seems really unwieldy. Instead I'd much rather see these e-2-e tests register an entirely new Pod user, and use that user for their tests, tearing it down afterwards - I'll just create a linked ticket for that!